src/components/Web3ConnectStatus/index.tsx

```diff
@@ -51,52 +52,62 @@ const Web3ConnectStatus = observer(props => {
     modalStore.toggleWalletModal();
   };
 
-  function getWeb3Status() {
-    console.debug('[GetWeb3Status]', {
-      account,
-      chainId: chainId,
-      error,
-    });
+  const toggleNetworkModal = () => {
+    modalStore.toggleNetworkModal();
+  };
+
+  function getWalletStatus() {
+    console.debug('[GetWalletStatus]', { account });
+    if (account) {
+      return (
+        <AccountButton onClick={toggleWalletModal}>
+          {shortenAddress(account)}
+        </AccountButton>
+      );
+    } else {
+      return (
+        <ConnectButton onClick={toggleWalletModal} active={true}>
+          {props.text}
+        </ConnectButton>
+      );
+    }
+  }
+
+  function getNetworkStatus() {
+    console.debug('[GetNetworkStatus]', { chainId, error });
     // Wrong network
-    if (account && chainId && !isChainIdSupported(chainId)) {
+    if ((chainId && !isChainIdSupported(chainId)) || error) {
       return (
-        <WrongNetworkButton onClick={toggleWalletModal}>
+        <WrongNetworkButton onClick={toggleNetworkModal}>
           Wrong Network
         </WrongNetworkButton>
       );
-    } else if (account) {
+    } else if (chainId) {
       return (
         <div style={{ display: 'flex' }}>
-          <AccountButton onClick={toggleWalletModal}>
-            {shortenAddress(account)}
-          </AccountButton>
           <AccountButton
-            onClick={toggleWalletModal}
+            onClick={toggleNetworkModal}
             style={{ fontSize: '14px' }}

```

Comment:
may be you would like to extend this AccountButton and remove this inline style?